### PR TITLE
Restore shelveset metadata and cleanup loc strings.

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: src/Agent.Worker/Build/TeeTFCommandManager.cs [hunk 1]
```diff
@@ -78,12 +78,12 @@ public string ResolvePath(string workspace, string serverPath)
...
 
-        public async Task ShelveAsync(string directory, string shelveset, string commentFile)
+        public async Task ShelveAsync(string shelveset, string directory, string commentFile)
         {
-            ArgUtil.NotNullOrEmpty(directory, nameof(directory));
             ArgUtil.NotNullOrEmpty(shelveset, nameof(shelveset));
+            ArgUtil.NotNullOrEmpty(directory, nameof(directory));
             ArgUtil.NotNullOrEmpty(commentFile, nameof(commentFile));
-            await RunPorcelainCommandAsync(FormatFlags.OmitCollectionUrl, "shelve", "-replace", "-recursive", $"-comment:@{commentFile}", shelveset, directory);
+            await RunPorcelainCommandAsync(FormatFlags.OmitCollectionUrl, "shelve", "-saved", "-replace", "-recursive", $"-comment:@{commentFile}", shelveset, directory);
```
Comment (ericsciple) [inline src/Agent.Worker/Build/TeeTFCommandManager.cs]: I was wrong! Adding this "-saved" arg fixes work item association for gated. Alex helped me figure it out.

File: src/Agent.Worker/Build/TeeTFCommandManager.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ public interface ITeeTFCommandManager : IAgentService
         Task EulaAsync();
         Task GetAsync(string version, string directory);
         string ResolvePath(string workspace, string serverPath);
-        Task ShelveAsync(string directory, string shelveset, string commentFile);
+        Task ShelveAsync(string shelveset, string directory, string commentFile);
```
Comment (ericsciple) [inline src/Agent.Worker/Build/TeeTFCommandManager.cs]: I flipped the order to be consistent with the command.

File: src/Agent.Worker/Build/TfsVCTeeSourceProvider.cs [hunk 3]
```diff
@@ -95,13 +95,23 @@ public sealed class TfsVCTeeSourceProvider : SourceProvider, ISourceProvider
...
 
                 // Undo any pending changes.
-                // TODO: Manually delete pending adds since they do not get deleted on undo?
                 if (existingTeeWorkspace != null)
                 {
                     TeeStatus teeStatus = await tf.StatusAsync(workspaceName);
                     if (teeStatus?.PendingChanges?.Any() ?? false)
                     {
                         await tf.UndoAsync(sourcesDirectory);
+
```
Comment (ericsciple) [inline src/Agent.Worker/Build/TfsVCTeeSourceProvider.cs]: This is just nice to add.

File: src/Agent.Worker/Build/TeeTFCommandManager.cs [hunk 4]
```diff
@@ -110,7 +110,7 @@ public string ResolvePath(string workspace, string serverPath)
         public async Task<TeeStatus> StatusAsync(string workspace)
         {
             ArgUtil.NotNullOrEmpty(workspace, nameof(workspace));
-            string xml = await RunPorcelainCommandAsync("status", $"-workspace:{workspace}", "-recursive", "-format:xml");
+            string xml = await RunPorcelainCommandAsync("status", $"-workspace:{workspace}", "-recursive", "-nodetect", "-format:xml");
```
Comment (TingluoHuang) [inline src/Agent.Worker/Build/TeeTFCommandManager.cs]: please confirm this. :D

**Comment by TingluoHuang** [comment]
Looks good.

**Pull request merged**